gen/unmarshal.go

```diff
@@ -325,9 +355,22 @@ func (u *unmarshalGen) gMap(m *Map) {
 	u.msgs = append(u.msgs, resizemsgs...)
 
 	// loop and get key,value
+	last := randIdent()
+	lastSet := randIdent()
+	u.p.printf("\nvar %s %s; _ = %s", last, m.Key.TypeName(), last) // we might not use the sort if it's not defined
+	u.p.declare(lastSet, "bool")
+	u.p.printf("\n_ = %s", lastSet) // we might not use the flag
 	u.p.printf("\nfor %s > 0 {", sz)
 	u.p.printf("\nvar %s %s; var %s %s; %s--", m.Keyidx, m.Key.TypeName(), m.Validx, m.Value.TypeName(), sz)
 	next(u, m.Key)
+	if m.Key.LessFunction() != "" {
```

Comment:
maybe do not require less func for all types? since numbers, strings, arrays are comparable, byte slices can be generated with bytes.Compare if m.Key.TypeName() == []bytes - i.e. to make this feature more usable by requiring less work by a user.

Btw, it looks like I can call UnmarshalValidateMsg on a type without providing LessFunction, this code would not be generated and I might get no err even there is an ordering issue.

I guess we need to go one of the roads:
1. allow `UnmarshalValidateMsg` only for types with `LessFunction` provided.
2. have `UnmarshalValidateMsg` to generate own code by propagating `validate=true/false` flag down in generator methods: `UnmarshalMsg` would generate regular stuff as now without `if validate` overhead, and `UnmarshalValidateMsg` would generate code with `if validate` + err if less function not provided but needed.

Reply:
> maybe do not require less func for all types? since numbers, strings, arrays are comparable, byte slices can be generated with bytes.Compare if m.Key.TypeName() == []bytes - i.e. to make this feature more usable by requiring less work by a user.

I agree -- I can make it smarter


> I guess we need to go one of the roads:
> 
> 1. allow `UnmarshalValidateMsg` only for types with `LessFunction` provided.
> 2. have `UnmarshalValidateMsg` to generate own code by propagating `validate=true/false` flag down in generator methods: `UnmarshalMsg` would generate regular stuff as now without `if validate` overhead, and `UnmarshalValidateMsg` would generate code with `if validate` + err if less function not provided but needed.

These two don't seem mutually exclusive to me. The first one is a correctness issue and the second seems like a performance concern.

I could fix the issue by including the `if validate` statement as separate and then inside that block do another if check if we have the LessFn and return an error uncoditionally if LessFn is missing.